operator/operator_test.go

```diff
@@ -164,11 +164,17 @@ func initEnvironment(t *testing.T) *environment {
 }
 
 func (e *environment) WaitPhase() {
-	waitBlock := func() {
-		time.Sleep(blockTime * time.Second)
+	sub, err := e.operator.ethClient.SubscribeToBlocks()
+	if err != nil {
+		e.T.Fatal("subscribing to blocks: ", err)
 	}
-	for i := uint64(0); i < e.enclaveParameters.PhaseDuration; i++ {
-		waitBlock()
+	defer sub.Unsubscribe()
+
+	for block := range sub.Blocks() {
+		// - 1 to make sure that we are in the new phase.
+		if e.enclaveParameters.IsLastPhaseBlock(block.NumberU64() - 1) {
```

Comment:
How is `WaitPhase` used? It's confusing to see this `-1` because now you're waiting until the first block of a new phase has been mined, not when a phase starts.

Reply:
It did not work without the `-1`. Maybe @matthiasgeihs can tell us what it was actually meant for.

Reply:
I implemented `IsLastPhaseBlock` and it does
```go
// IsLastPhaseBlock tells whether this block is the last block of a phase.
func (p Parameters) IsLastPhaseBlock(blockNum uint64) bool {
	return ((blockNum - p.InitBlock) % p.PhaseDuration) == p.PhaseDuration-1
}
```
so from my understanding, it is `true` for the last block of any phase. I.e., if the last block is _mined_ then the phase is done.